src/solana/index.ts

```diff
@@ -340,50 +376,400 @@ export class Solana extends Blockchain {
+          authorizedPubkey: senderPublicKey,
+        });
+
+        instructions.push(...tx.instructions, ...deactivateTx.instructions);
+      }
+
+      for (const acc of accountsToDeactivate) {
+        const deactivateTx = StakeProgram.deactivate({
+          stakePubkey: acc.pubkey,
+          authorizedPubkey: senderPublicKey,
+        });
+
+        instructions.push(...deactivateTx.instructions);
+      }
+
+      // cast instructions to correct JSON Serialization
+      instructions = instructions.map((instruction) => {
+        return new TransactionInstruction(instruction);
+      });
+
+      const versionedTX = await this.prepareTransaction(
+        instructions,
+        senderPublicKey,
+        [],
+      );
+
+      await SetStats({
+        action: 'unstake',
+        address: sender,
+        amount: lamports / LAMPORTS_PER_SOL,
+        chain: CHAIN,
+        token: token,
+      });
+
+      return { result: versionedTX };
+    } catch (error) {
+      throw this.handleError('UNSTAKE_ERROR', error);
+    }
+  }
+
+  private async split(
+    authorityPublicKey: PublicKey,
+    lamports: number,
+    oldStakeAccountPubkey: PublicKey,
+    source: string,
+  ): Promise<[Transaction, PublicKey, Keypair[]]> {
+    // Format source to
+    const seed = this.formatSource(source);
+
+    const newStakeAccountPubkey = await PublicKey.createWithSeed(
+      authorityPublicKey,
+      seed,
+      StakeProgram.programId,
+    );
+
+    const splitStakeAccountTx = new Transaction().add(
+      StakeProgram.splitWithSeed({
+        stakePubkey: oldStakeAccountPubkey,
+        authorizedPubkey: authorityPublicKey,
+        splitStakePubkey: newStakeAccountPubkey,
+        basePubkey: authorityPublicKey,
+        seed: seed,
+        lamports: lamports,
+      }),
+    );
+
+    return [splitStakeAccountTx, newStakeAccountPubkey, []];
+  }
+
+  public async claim(
+    sender: string,
+  ): Promise<ApiResponse<VersionedTransaction>> {
+    try {
+      const delegations = await this.getDelegations(sender);
+
+      const stakeAccounts = delegations.result.map((delegationAcc) => {
+        return {
+          pubkey: delegationAcc.pubkey,
+          account: parsedAccountInfoToStakeAccount(delegationAcc.account),
+        };
+      });
+
+      const epochInfo = await this.connection.getEpochInfo();
+      const tm = this.timestampInSec();
+
+      let totalClaimableStake = new BigNumber(0);
+      const deactivatedStakeAccounts = stakeAccounts.filter((acc) => {
+        const isDeactivated =
+          !isLockupInForce(acc.account.data.info.meta, epochInfo.epoch, tm) &&
+          stakeAccountState(acc.account.data, epochInfo.epoch) ===
+            StakeState.Deactivated;
+
+        if (acc.account.data.info.stake != null && isDeactivated) {
+          totalClaimableStake = totalClaimableStake.plus(
+            acc.account.data.info.stake.delegation.stake,
+          );
+        }
+
+        return isDeactivated;
+      });
```

Comment:
```suggestion
const deactivatedStakeAccounts = stakeAccounts.filter((acc) => {
  const { data } = acc.account;
  const { info } = data;

  const isDeactivated =
    !isLockupInForce(info.meta, epochInfo.epoch, tm) &&
    stakeAccountState(data, epochInfo.epoch) === StakeState.Deactivated;

  if (info.stake && isDeactivated) {
    totalClaimableStake = totalClaimableStake.plus(info.stake.delegation.stake);
  }

  return isDeactivated;
```

use object destructuring